web/modules/custom/hoeringsportal_content_blocks/src/Plugin/Block/MeetingWarning.php

```diff
@@ -12,16 +15,61 @@
  *   admin_label = @Translation("Node public meeting warning"),
  * )
  */
-class MeetingWarning extends BlockBase {
+class MeetingWarning extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The node the block appears on.
+   *
+   * @var \Drupal\Core\Routing\CurrentRouteMatch
+   */
+  protected $currentNode;
+
+  /**
+   * Block constructor.
+   *
+   * @param array $configuration
+   *   Block configuration.
+   * @param string $plugin_id
+   *   Block plugin id.
+   * @param mixed $plugin_definition
+   *   Block plugin definition.
+   * @param \Drupal\Core\Routing\CurrentRouteMatch $routeMatch
+   *   The route match.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, CurrentRouteMatch $routeMatch) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->currentNode = $routeMatch->getParameter('node');
+  }
+
+  /**
+   * Create block.
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   Container interface.
+   * @param array $configuration
+   *   Block configuration.
+   * @param string $plugin_id
+   *   Block plugin id.
+   * @param mixed $plugin_definition
+   *   Block plugin definition.
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('current_route_match')
+    );
+  }
 
   /**
    * {@inheritdoc}
    */
   public function build() {
-    // Get current node.
-    $node = \Drupal::routeMatch()->getParameter('node');
-    if (isset($node) && $node->field_content_state->value) {
-      $config['deadline_passed'] = $node->field_content_state->value;
+    if (isset($this->currentNode) && $this->currentNode->field_content_state->value) {
```

Comment:
You should get the current from the routeMatch service here, e.g. `$node = $this->routeMatch->getParameter('node')`.